### PR TITLE
Add config to chained validator

### PR DESCRIPTION
**Pull request by Richard87**

### CODE REVIEW

File: api/utils/token/chained_validator.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (principal TokenPrincipal, err error) {
+	for index, validator := range v.validators {
+		principal, err = validator.ValidateToken(ctx, token)
+		if err == nil {
+			return principal, nil
+		} else if index == len(v.validators)-1 {
+			return nil, err
+		}
+	}
```
Comment (satr) [inline api/utils/token/chained_validator.go:25]: Optional suggestion - to make it easier to read
```suggestion
		if principal, err = validator.ValidateToken(ctx, token); err == nil {
			return principal, nil
		} else if index == len(v.validators)-1 {
			return nil, err
		}
```
or
```suggestion
		principal, err = validator.ValidateToken(ctx, token)
		if err == nil {
			return principal, nil
		} 
		if index == len(v.validators)-1 {
			return nil, err
		}
```
or 
```suggestion
	var principal TokenPrincipal
	var err error
	for _, validator := range v.validators {
		if principal, err = validator.ValidateToken(ctx, token); err == nil {
			return principal, nil
		}
	}
	if err != nil {
	  return nil, err
	}
```

File: api/utils/token/chained_validator.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+type ChainedValidator struct{ validators []ValidatorInterface }
+
+var _ ValidatorInterface = &ChainedValidator{}
+var errNoIssuersFound = errors.New("no issuers found")
+
+func NewChainedValidator(validators ...ValidatorInterface) *ChainedValidator {
+	return &ChainedValidator{validators}
+}
+
+func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (principal TokenPrincipal, err error) {
```
Comment (satr) [inline api/utils/token/chained_validator.go:17]: No need for names in returned values - method returns them explicitly, like `return principal, nil` or `return nil, err`

File: main.go [hunk 3]
```diff
@@ -77,18 +76,18 @@ func initializeServer(c config.Config) *http.Server {
 }
 
 func initializeTokenValidator(c config.Config) token.ValidatorInterface {
-	issuerUrl, err := url.Parse(c.OidcIssuer)
+	azureValidator, err := token.NewValidator(c.AzureOidc.Issuer, c.AzureOidc.Audience)
 	if err != nil {
-		log.Fatal().Err(err).Msg("Error parsing issuer url")
+		log.Fatal().Err(err).Msg("Error creating JWT validator")
```
Comment (satr) [inline main.go:81]:
```suggestion
		log.Fatal().Err(err).Msg("Error creating JWT Azure OIDC validator")
```

File: main.go [hunk 4]
```diff
@@ -77,18 +76,18 @@ func initializeServer(c config.Config) *http.Server {
...
 	}
 
-	// Set up the validator.
-	// jwtValidator, err := token.NewValidator(issuerUrl, c.OidcAudience)
-	jwtValidator, err := token.NewUncheckedValidator(issuerUrl, c.OidcAudience)
+	kubernetesValidator, err := token.NewValidator(c.KubernetesOidc.Issuer, c.KubernetesOidc.Audience)
 	if err != nil {
 		log.Fatal().Err(err).Msg("Error creating JWT validator")
```
Comment (satr) [inline main.go:86]:
```suggestion
		log.Fatal().Err(err).Msg("Error creating JWT Kubernetes OIDC validator")
```

**Review by satr — DISMISSED** [review]

File: api/utils/token/chained_validator.go (hunk 2 above)
Comment (Richard87) [reply api/utils/token/chained_validator.go:17]: agreed, returning a list of errors if not found :)

File: api/utils/token/chained_validator.go (hunk 1 above)
Comment (Richard87) [reply api/utils/token/chained_validator.go:25]: fixed with:

```golang
func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (TokenPrincipal, error) {
	var errs []error

	for _, validator := range v.validators {
		principal, err := validator.ValidateToken(ctx, token)
		if principal != nil {
			return principal, nil
		}
		errs = append(errs, err)
	}

	return nil, fmt.Errorf("%w: %v", errNoIssuersFound, errors.Join(errs...))
}
```

**Review by satr — DISMISSED** [review]

**Review by satr — APPROVED** [review]

**Pull request merged**